src/core/src/handle.rs

```diff
@@ -67,43 +128,82 @@ pub struct Shader<R: Resources>(Arc<R::Shader>);
 
 /// Program Handle
 #[derive(Clone, Debug)]
-pub struct Program<R: Resources>(Arc<R::Program>, shade::ProgramInfo);
+pub struct Program<R: Resources>(Arc<ProgramData<R>>);
```

Comment:
This is going to greatly help the performance of program handles cloning, since `ProgramInfo` is rather heavy.